vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -35,6 +47,21 @@ def _filter_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
     return series.isin(value)
 
 
+def _filter_date_between(series: pd.Series, value: List[str]) -> pd.Series:
+    series = pd.to_datetime(series)
+    start_date, end_date = value
+
+    return series.between(start_date, end_date, inclusive="both")
+
+
+def _filter_date_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
+    series = pd.to_datetime(series)
+    return series.isin(value)
+
+
+FILTER_FUNCTIONS = {RangeSlider: _filter_between, DateRangePicker: _filter_date_between, DatePicker: _filter_date_isin}
```

Comment:
I haven't tried it out, so it might not work, but worth to give it a shot.

If instead of providing the datetime values as strings and then converting them here, what if you provide them directly as datetime objects? I think you wouldn't need these different filter functions then, but can then just re-use the existing ones `_filter_is_in` and `_filter_between`

The datetime object conversion should happen in the pydantic models before then

Reply:
Did this not work @nadijagraca ?